packages/lesswrong/components/comments/CommentsNewForm.tsx

```diff
@@ -284,8 +298,51 @@ const CommentsNewForm = ({
     if (formDisabledDueToRateLimit) {
       submitBtnProps.disabled = true
     }
-    
+
+    const { DialogueCheckBox } = Components
```

Comment:
It really feels like this entire thing should be its own component - there's a whole bunch of awkward logic trying to deal with the fact that e.g. `parentComment` might not exist, which would be easily solved by pulling the whole thing out into its own component and having `parentComment` be a required prop.  (And then you wouldn't render the component if there was no `parentComment`.)

It's probably not worth blocking a deploy but I would definitely follow up with that.

Reply:
(also `currentUser`)